pkg/cluster/kubernetes/upgrade_checks_test.go

```diff
@@ -0,0 +1,209 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package kubernetes_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/cosi-project/runtime/pkg/state/impl/inmem"
+	"github.com/cosi-project/runtime/pkg/state/impl/namespaced"
+	"github.com/siderolabs/go-retry/retry"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/siderolabs/talos/pkg/cluster/kubernetes"
+	"github.com/siderolabs/talos/pkg/machinery/resources/k8s"
+)
+
+type UpgradeCheckSuite struct {
+	suite.Suite
+}
+
+func TestUpgradeChecksSuite(t *testing.T) {
+	suite.Run(t, &UpgradeCheckSuite{})
+}
+
+func (suite *UpgradeCheckSuite) TestK8sComponentRemovedItemsNoError() {
+	ctx, ctxCancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer ctxCancel()
+
+	resourceState := state.WrapCore(namespaced.NewState(inmem.Build))
+
+	for _, id := range []string{k8s.APIServerID, k8s.ControllerManagerID, k8s.SchedulerID} {
+		cfg := k8s.NewStaticPod(k8s.NamespaceName, id)
+		cfg.TypedSpec().Pod = map[string]interface{}{
+			"spec": map[string]interface{}{
+				"containers": []interface{}{
+					map[string]interface{}{
+						"command": []string{
+							fmt.Sprintf("/usr/local/bin/%s", id),
+						},
+					},
+				},
+			},
+		}
+
+		suite.Require().NoError(resourceState.Create(ctx, cfg))
+
+		suite.Assert().NoError(
```

Comment:
I believe this check is not needed, as resource is immediately available after creation (same for the check on line 175)

Reply:
for some reason I was thinking this was checking the staticpodstatus resource :facepalm: